x-pack/plugins/fleet/common/constants/routes.ts

```diff
@@ -121,6 +121,7 @@ export const AGENT_API_ROUTES = {
   UPGRADE_PATTERN: `${API_ROOT}/agents/{agentId}/upgrade`,
   BULK_UPGRADE_PATTERN: `${API_ROOT}/agents/bulk_upgrade`,
   CURRENT_UPGRADES_PATTERN: `${API_ROOT}/agents/current_upgrades`,
+  INTERNAL_LIST_PATTERN: `${INTERNAL_ROOT}/agents`, // for testing
```

Comment:
Could we query the agents index directly using the `esClient` we have in API integration tests rather than standing up this internal endpoint?

Reply:
The goal here was to test the logic of the PIT + batching logic in `getAllAgentsByKuery` (query max 10k at a time), this was the easiest way to set a smaller batch size to test (it would be hard to put 10k documents in test snapshot).
Having said that, the batching in `getAllAgentsByKuery` is not being called anymore by actions, since the bulk action handlers are calling `processAgentsInBatches` now. So I am okay to remove this internal route.